manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+		svc.Register(validUser)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
+		validChannel.ID = chanID
+
+		cases := []struct {
+			desc   string
+			chanID string
+			auth   string
+			status int
+		}{
+			{"remove existing channel", validChannel.ID, validUser.Email, http.StatusNoContent},
+			{"try to remove non-existent channel", validChannel.ID, validUser.Email, http.StatusNoContent},
+			{"try to remove channel with invalid id", "123e4567-e89b-12d3-a456-000000000042", validUser.Email, http.StatusNoContent},
+			{"try to remove channel with invalid token", validChannel.ID, invalidUser.Email, http.StatusForbidden},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodDelete, fmt.Sprintf("%s/channels/%s", url, tc.chanID), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestConnect(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clientID, _ := svc.AddClient(validUser.Email, validClient)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
+
+		cases := []struct {
+			desc     string
+			chanID   string
+			clientID string
+			auth     string
+			status   int
+		}{
+			{"connect existing client to existing channel", chanID, clientID, validUser.Email, http.StatusOK},
+			{"try to connect existing client to non-existent channel", "123e4567-e89b-12d3-a456-000000000042", clientID, validUser.Email, http.StatusNotFound},
+			{"try to connect client with invalid id to channel", chanID, "1", validUser.Email, http.StatusNotFound},
+			{"try to connect client to channel with invalid id", "1", clientID, validUser.Email, http.StatusNotFound},
+			{"try to connect existing client to existing channel with invalid token", chanID, clientID, invalidUser.Email, http.StatusForbidden},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodPut, fmt.Sprintf("%s/channels/%s/clients/%s", url, tc.chanID, tc.clientID), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestDisconnnect(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clientID, _ := svc.AddClient(validUser.Email, validClient)
+		chanID, _ := svc.CreateChannel(validUser.Email, validChannel)
+		svc.Connect(validUser.Email, chanID, clientID)
+
+		cases := []struct {
+			desc     string
+			chanID   string
+			clientID string
+			auth     string
+			status   int
+		}{
+			{"disconnect connected client from channel", chanID, clientID, validUser.Email, http.StatusNoContent},
```

Comment:
Add tests that try to disconnect channel from user A from client from user B and vice versa. This is an upgrade to the invalid token test :)